### DISCUSSION TITLE
BO: Sort items/products for lists>300 not possible

### DISCUSSION THREAD

**Discussion by Shoprunners** [PrestaShop/PrestaShop#33441] [Ideas]
If you would like to sort i.e. products in a category and you have more than 300 products, it is currently not possible to bring the products from pos > 300 to a position <= 300. 

**To Reproduce**
take a category with more than 300 products assigned. Show 300 products at once and go to page 2. Now try to move a product from that page to pos #1. Not possible

Possible (quick) solution: Maybe change or add a pagination segment to sth, which is not matching with 300, sth. like 125. At least you will be able to do a workaround. Maybe you can add a mass edit function like "move to top".

PrestaShop version: 1.7.6.1

**Comment by khouloudbelguith** [comment]
Hi @Shoprunners,

Thanks for your report.
Yes, I have the same behavior with PS1.7.5.2 & PS1.7.6.1
https://drive.google.com/file/d/1HMXkU2OlfzMYLcDTB5iTmIcvE8friSRN/view
Ping @PrestaShop/prestashop-product-team what do you think of his improvement?

Thanks!

**Comment by psyray** [comment]
I think it's not an improvement but a bug.
We could type a position to move an object on top, not only drag and drop.
So add a text input for the position number with a form to save position could solve the issue

**Comment by marionf** [comment]
@TristanLDD What do you think ?

**Comment by TristanLDD** [comment]
Thank you @Shoprunners for the suggestion, I think we have to make a mass edit option that allows you to moove a selection of products from initial position range to another : 
![Capture d’écran 2019-09-17 à 11 47 42](https://user-images.githubusercontent.com/27484094/65031346-67dab980-d941-11e9-89f3-2ec941b355fa.png)

**Comment by fire2** [comment]
Sorting of products in pages > 1 also doesn't work, e.g, moving product from position 302 to 301, not just from page 2 to page 1.

**Comment by haveyougotanypets** [comment]
The quick fix @Shoprunners suggest of changing the default pagination options to say 120 can be achieved by changing the file 

`src/Adapter/Product/AdminProductDataProvider.php`

function `getPaginationLimitChoices`

`$paginationLimitChoices = [20, 50, 100];` to `$paginationLimitChoices = [20, 50, 100, 120];`

Thanks to  [this forum post](https://www.prestashop.com/forums/topic/1001259-back-office-pagination-prestashop-1761/) for tracking down the code

Note: You could also be more aggressive and add in the 1000 option that the `getMemoryLimit()` method further down the function that is holding back from you ;)

**Comment by TechhDan** [comment]
I have the same issue with a client who has many categories. They are unable to move the position in a sub-category from position > 100 to position <= 100. I hope this can be considered for next PrestaShop updates.

For the moment I have manually changed ```AdminProductController->paginationAction``` I've added 1000 to the ```$limitChoices``` array on 1.7.8.8.

**Comment by Shoprunners** [comment]
Well, I don't understand why my suggestion was not approved. Just a 3 digit change. with no risk. Not the perfect solution, but implementing mass updates and extra function does take too long and I guess the workload is already very high with other important things.

**Comment by kpodemski** [comment]
Hello everyone. Following our new approach to feature requests, I converted this Issue to the GitHub Discussion so that the community can discuss the best possible ways to handle this problem and maybe, I hope, organize together to work towards fixing it in the core. If you are motivated to do so, we will be happy to help.
